src/screens/CreatePost/index.js

```diff
@@ -435,54 +435,57 @@ const CreatePost = () => {
       return true;
     }
 
-    const topicsToPost = _.union(initialTopic, listTopic);
-    const data = {
-      message,
-      topics: topicsToPost,
-      verb: isPollShown ? 'poll' : 'tweet',
-      feedGroup: 'main_feed',
-      privacy: listPrivacy[0].key,
-      anonimity: typeUser,
-      location: renderLocationString(geoList[geoSelect]),
-      location_id: locationId,
-      duration_feed: postExpired[expiredSelect].value,
-      images_url: dataImage,
-      tagUsers: checkTaggingUser(),
-      is_photo_uploaded: true
-    };
-
-    if (isPollShown) {
-      data.polls = getReducedPoll();
-      data.pollsduration = selectedTime;
-      data.multiplechoice = isPollMultipleChoice;
-    }
-
-    if (typeUser) {
-      data.anon_user_info = {
-        color_name: anonUserInfo?.colorName,
-        color_code: anonUserInfo?.colorCode,
-        emoji_name: anonUserInfo?.emojiName,
-        emoji_code: anonUserInfo?.emojiCode
-      };
-    }
-
-    setDurationId(JSON.stringify(expiredSelect));
-    if (!isInCreatePostTopicScreen) {
-      setLocationId(JSON.stringify(geoSelect));
-      setPrivacyId(JSON.stringify(0));
-    }
-
     const maxRetries = 5;
-
     const attemptPost = async (retryCount) => {
       try {
+        const topicsToPost = _.union(initialTopic, listTopic);
+        console.log('message', message);
+        const data = {
+          message,
+          topics: topicsToPost,
+          verb: isPollShown ? 'poll' : 'tweet',
+          feedGroup: 'main_feed',
+          privacy: listPrivacy[0].key,
+          anonimity: typeUser,
+          location: renderLocationString(geoList[geoSelect]),
+          location_id: locationId,
+          duration_feed: postExpired[expiredSelect].value,
+          images_url: dataImage,
+          tagUsers: checkTaggingUser(),
+          is_photo_uploaded: true
+        };
+
+        if (isPollShown) {
+          data.polls = getReducedPoll();
+          data.pollsduration = selectedTime;
+          data.multiplechoice = isPollMultipleChoice;
+        }
+
+        if (typeUser) {
+          data.anon_user_info = {
+            color_name: anonUserInfo?.colorName,
+            color_code: anonUserInfo?.colorCode,
+            emoji_name: anonUserInfo?.emojiName,
+            emoji_code: anonUserInfo?.emojiCode
+          };
+        }
+
+        setDurationId(JSON.stringify(expiredSelect));
+        if (!isInCreatePostTopicScreen) {
+          setLocationId(JSON.stringify(geoSelect));
+          setPrivacyId(JSON.stringify(0));
+        }
+
         const post = await createPost(data);
+        if (params.onRefresh && typeof params.onRefresh === 'function') {
+          params.onRefresh();
+        }
+        setLoadingPost(false);
         if (post.code === 200) {
           showMessage({
             message: StringConstant.createPostDone,
             type: 'success'
           });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `attemptPost` function has been refactored to include retry logic. This is a good practice for handling network requests, as it can help mitigate transient issues. However, the current implementation does not actually implement any retry logic. The `retryCount` parameter is never incremented and the function is not recursively called in case of failure. To implement retry logic, you could use a loop or recursion to call `attemptPost` again if the post fails and the `retryCount` is less than `maxRetries`. Also, consider adding a delay before retrying the request to give the network some time to recover.

```diff
+      const attemptPost = async (retryCount = 0) => {
         try {
           // ...
           const post = await createPost(data);
           // ...
         } catch (e) {
           if (__DEV__) {
             console.log('CreatePost : ', e);
             console.warn('retryCount', retryCount);
           }
+          if (retryCount < maxRetries) {
+            setTimeout(() => attemptPost(retryCount + 1), 1000); // wait for 1 second before retrying
+          } else {
             showMessage({
               message: 'Failed to post. Please check your internet connection.',
               type: 'danger'
             });
             setLoadingPost(false);
+          }
         }
       };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->